lambda-hot-reloading/javascript-terraform-layers/main.tf

```diff
@@ -0,0 +1,71 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.0"
+    }
+  }
+}
+
+# Configure the AWS Provider
+provider "aws" {
+  region = "us-east-1"
+}
+
+resource "aws_iam_role" "iam_for_lambda" {
+  name = "iam_for_lambda"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_lambda_layer_version" "test_layer" {
+  layer_name = "hot_reload_layer"
+  compatible_runtimes = ["nodejs16.x"]
+
+  s3_bucket = "hot-reload"
+  s3_key    = "${abspath(path.root)}/layer_src"
+
+}
+
+resource "aws_lambda_function" "test_lambda" {
+  # If the file is not in the current working directory you will need to include a
+  # path.module in the filename.
+  function_name = "hotreloadlambda"
+  role          = aws_iam_role.iam_for_lambda.arn
+  handler       = "index.handler"
+
+
+  s3_bucket = "hot-reload"
+  s3_key    = "${abspath(path.root)}/lambda_src"
+  layers = [aws_lambda_layer_version.test_layer.arn]
+
+  runtime = "nodejs16.x"
```

Comment:
I guess this wouldn't work with nodejs18.x out of the box right now?

Reply:
Probably not, due to the handler code.